core/src/drawable.rs

```diff
@@ -106,58 +158,111 @@ pub trait Drawable {
         D: DrawTarget<Color = Self::Color>;
 }
 
-/// A single pixel.
-///
-/// `Pixel` objects are used to specify the position and color of drawn pixels.
-///
-/// # Examples
-///
-/// The [`Drawable`] trait is implemented for `Pixel` which allows single pixels
-/// to be drawn to a [`DrawTarget`]:
-/// ```
-/// use embedded_graphics::{pixelcolor::BinaryColor, prelude::*};
-/// # use embedded_graphics::mock_display::MockDisplay;
-/// # let mut display = MockDisplay::new();
-///
-/// Pixel(Point::new(1, 2), BinaryColor::On).draw(&mut display)?;
-/// # Ok::<(), core::convert::Infallible>(())
-/// ```
-///
-/// Iterators with `Pixel` items can also be drawn:
-///
-/// ```
-/// use embedded_graphics::{pixelcolor::BinaryColor, prelude::*};
-/// # use embedded_graphics::mock_display::MockDisplay;
-/// # let mut display = MockDisplay::new();
-///
-/// (0..32)
-///     .map(|i| Pixel(Point::new(i, i * 2), BinaryColor::On))
-///     .draw(&mut display)?;
-/// # Ok::<(), core::convert::Infallible>(())
-/// ```
-///
-/// [`DrawTarget`]: crate::draw_target::DrawTarget
-#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug, Default)]
-#[cfg_attr(feature = "defmt", derive(::defmt::Format))]
-pub struct Pixel<C>(pub Point, pub C)
-where
-    C: PixelColor;
+impl<T: Drawable> Drawable for &T {
```

Comment:
What about making drawing work on slices?

```rust
// add ?Sized
impl<T: Drawable + ?Sized> Drawable for &T {...}

// add this impl
impl<T: Drawable> Drawable for [T] {...}
```

Reply:
`Drawable`, unfortunately, cannot be implemented for slices without either:
- changing the trait (did not want to introduce any breaking changes in this PR) 
- having the associated `Output` type of the `Drawable` trait not be meaningful, or lossy, when implemented on slices. 

This is the same for the lack of slice implementations on other traits which return a success value from one or more of their trait methods. 

With respect to the `?Sized` addition, I'd be fine to add this for the traits in which it makes sense and it feels like a worthwhile change